tracer/test/Datadog.Trace.TestHelpers/ProfilerHelper.cs

```diff
@@ -30,6 +36,61 @@ public class ProfilerHelper
             // clear all relevant environment variables to start with a clean slate
             EnvironmentHelper.ClearProfilerEnvironmentVariables();
 
+            // this is nasty, but it's the only way I could find to force
+            // a .NET Framework exe to run in 32 bit if required
+            if (EnvironmentTools.IsWindows()
+             && !EnvironmentHelper.IsCoreClr()
+             && !EnvironmentTools.IsTestTarget64BitProcess()
```

Comment:
Are we using the same sample.exe for both architectures? If so, don't we want to run this code in both cases, once to set the `32BITREQ` flag and and one to remove it (in whatever order)? We also need to make sure they don't run in parallel.

Reply:
As mentioned [here](https://github.com/DataDog/dd-trace-dotnet/pull/3632#discussion_r1064458935), on the one hand, yes, you're completely right, we are using the same exe for both architectures, and we're not currently clearing the flag between runs.

However, that's not an issue for our current build, because:

- We don't run x86/x64 in parallel in the same build (locally, or in CI)
- Currently we don't run x86/x64 on the same _machine_
- When running in VS/Rider, we will always default to x64 (unless you go out of your way to set the `TargetFramework` env var
- When running with nuke (e.g `./build.ps1 BuildAndRunWindowsIntegrationTests -Framework net462 -TargetPlatform x86`) you need to explicitly choose one platform or the other. Running the command again with the other platform rebuilds the sample, essentially resetting the flag.

I'm not averse to _always_ setting the flag&mdash;not setting it is essentially a micro-optimisation based on our current build/solution&mdash. If consensus is to set it, we simply need to remove this one line 🙂